### DISCUSSION TITLE
Finding mode volume with python interface

### DISCUSSION THREAD

**Discussion by Dev-linkon** [NanoComp/meep#2138] [Q&A] [answered]
Hi there,

I want to simulate a photonic crystal cavity wthere the ratio of quality factor and the mode volume (Q/V) is very important. By running the simulations for resonant modes with 'Harminv', I am able to obtain the Q but I can not obtain any information about V. 

Kindly help me with the Python script to find out the mode volume (V).

Thanks for reading

**Comment by stevengj** [comment, marked as answer]
Realize that Q/V is simply an approximation for the LDOS at the point of maximum field intensity, so I would generally recommend computing LDOS instead.  Meep has a built-in `dft_ldos` feature to help with that, and the [Meep LDOS tutorial](https://meep.readthedocs.io/en/latest/Scheme_Tutorials/Local_Density_of_States/) has an explicit comparison with Q/V.

See [this textbook chapter](https://arxiv.org/abs/1301.5366), section 4.4.6, and also [this paper](http://doi.org/10.1364/OE.21.030812) on cavity optimization.

Alternatively, if you want to compute the modal volume directly, then you need the E field pattern of the mode.   The easiest way to get this, after you have already run harminv to find the resonant mode frequencies, is to run a new simulation with a narrow-band source around the resonant frequency (so that you excite only the mode).  After the source is done, the E field should be approximately that of the mode, and then you can use the modal-volume formula with this.
